test/e2e/operator.go

```diff
@@ -321,3 +325,91 @@ var _ = XDescribe("ARO Operator - MachineSet Controller", func() {
 		}
 	})
 })
+
+var _ = Describe("ARO Operator - Azure Subnet Reconciler", func() {
+	var vnetName, location, resourceGroup string
+	var subnetsToReconcile []string
+	var dummynsg mgmtnetwork.SecurityGroup
+	ctx := context.Background()
+
+	// TODO (robryan) rm this func once default to on https://github.com/Azure/ARO-RP/issues/1735
+	enableReconcileSubnet := func() {
+		instance, err := clients.AROClusters.AroV1alpha1().Clusters().Get(ctx, arov1alpha1.SingletonClusterName, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		if !instance.Spec.Features.ReconcileSubnets {
+			instance.Spec.Features.ReconcileSubnets = true
+			_, err = clients.AROClusters.AroV1alpha1().Clusters().Update(ctx, instance, metav1.UpdateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+		}
+	}
+	// Gathers vnet name, resource group, location, and adds master/worker subnets to list to reconcile.
+	gatherNetworkInfo := func() {
+		oc, err := clients.OpenshiftClustersv20200430.Get(ctx, vnetResourceGroup, clusterName)
+		Expect(err).NotTo(HaveOccurred())
+		location = *oc.Location
+
+		vnet, masterSubnet, err := subnet.Split(*oc.OpenShiftClusterProperties.MasterProfile.SubnetID)
+		Expect(err).NotTo(HaveOccurred())
+		_, workerSubnet, err := subnet.Split((*(*oc.OpenShiftClusterProperties.WorkerProfiles)[0].SubnetID))
+		Expect(err).NotTo(HaveOccurred())
+
+		subnetsToReconcile = append(subnetsToReconcile, masterSubnet, workerSubnet)
```

Comment:
Hm... Are you sure about `append` here? If we add more than one test into this `Describe` block the list will keep growing with duplicates. Or am I missing something?